jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletMultiPartFormData.java

```diff
@@ -49,106 +50,103 @@
 public class ServletMultiPartFormData
 {
     /**
-     * <p>Parses the request content assuming it is a multipart content,
-     * and returns a {@link Parts} objects that can be used to access
-     * individual {@link Part}s.</p>
-     *
-     * @param request the HTTP request with multipart content
-     * @return a {@link Parts} object to access the individual {@link Part}s
-     * @throws IOException if reading the request content fails
-     * @see org.eclipse.jetty.server.handler.DelayedHandler
+     * Get future {@link ServletMultiPartFormData.Parts} from a servlet request.
+     * @param servletRequest A servlet request
+     * @return A future {@link ServletMultiPartFormData.Parts}, which may have already been created and/or completed.
+     * @see #from(ServletRequest, String)
      */
-    public static Parts from(ServletApiRequest request) throws IOException
+    public static CompletableFuture<Parts> from(ServletRequest servletRequest)
```

Comment:
Should not this be `HttpServletRequest`? As parts are HTTP things.

Reply:
It will have to be-a HttpServletRequest, but just going for the lowest common denominator as elsewhere in the code.